typing/ctype.ml

```diff
@@ -2723,9 +2680,7 @@ and unify3 env t1 t1' t2 t2' =
     begin match !umode with
     | Expression ->
         occur !env t1' t2';
-        if is_self_type d1 (* PR#7711: do not abbreviate self type *)
-        then link_type t1' t2'
-        else link_type t1' t2
+        link_type t1' t2
```

Comment:
Does it mean that you solved the problem in #7711 ?

Reply:
Yes. There are now no special syntactic restrictions on self types.